<a href="https://colab.research.google.com/github/Ilicio/TensorFlow/blob/main/beginner.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#TensorFlow 2 início rápido para iniciantes




Esta breve introdução usa [Keras] (https://www.tensorflow.org/guide/keras/overview) para:

1. Criar uma rede neural que classifique imagens.
2. Treinar essa rede neural.
3. E, finalmente, avaliar a precisão do modelo.






Baixe e instale o pacote TensorFlow 2. Importe o TensorFlow para o seu programa:

In [1]:
from __future__ import absolute_import, division, print_function, unicode_literals

# Install TensorFlow
try:
  # %tensorflow_version only exists in Colab.
  %tensorflow_version 2.x
except Exception:
  pass

import tensorflow as tf

Carregue e prepare o [conjunto de dados MNIST] (http://yann.lecun.com/exdb/mnist/). 
Converta as amostras de números inteiros em números de ponto flutuante:

In [2]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11501568/11490434 [==============================] - 0s 0us/step


Crie o modelo tf.keras.Sequential empilhando camadas. Escolha uma função otimizadora e de perda para treinamento:

In [3]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Treine e avalie o modelo:

In [4]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)

Epoch 1/5
1875/1875 [==============================] - 5s 2ms/step - loss: 0.2937 - accuracy: 0.9150
Epoch 2/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.1443 - accuracy: 0.9581
Epoch 3/5
1875/1875 [==============================] - 5s 2ms/step - loss: 0.1085 - accuracy: 0.9675
Epoch 4/5
1875/1875 [==============================] - 5s 2ms/step - loss: 0.0886 - accuracy: 0.9719
Epoch 5/5
1875/1875 [==============================] - 4s 2ms/step - loss: 0.0767 - accuracy: 0.9758
313/313 - 1s - loss: 0.0751 - accuracy: 0.9761 - 517ms/epoch - 2ms/step


[0.07507722824811935, 0.9761000275611877]

O classificador de imagem agora é treinado para ~98% de acurácia neste conjunto de dados. Para saber mais, leia os [tutoriais do TensorFlow] (https://www.tensorflow.org/tutorials/).